lib/linalg/BasisWriter.cpp

```diff
@@ -81,13 +80,11 @@ BasisWriter::writeBasis(const std::string& kind)
     sprintf(tmp, "time_%06d", d_num_intervals_written);
```

Comment:
In this class, `d_num_intervals_written` is output to the file in the `basis` case, not the `snapshot` case, and it simply counts the number of times a `basis` has been written. On the other hand, the class `BasisGenerator` has another concept of "intervals", which I assume should be related to the interval concept in `BasisWriter`. As far as I know, there is no use case of the interval concept in practice, although the code was written with the intent of some use case. I guess the idea was that after a certain number of intervals is reached in `BasisGenerator`, the `BasisWriter` should write the basis, and the `BasisGenerator` should go to the next interval. We don't use these classes in this way, and this design is not tested, so it may even be broken. In any case, it may be best to remove the interval concept from both classes.

Reply:
I've looked through the examples, and indeed there is no case of using more than 1 interval. Does this mean that time-windowing has been implemented differently, not using this `interval` concept?

Reply:
Yes, we do have a time-windowing method that is implemented in applications and examples (e.g. Laghos), but it is not part of the libROM library. Time-windowing always uses 1 "interval" per time window. The determination to end a window is made in the application, not in libROM, since it should be problem-dependent. We do not use a general way to define windows that would be suitable for the library in general. When a window is ended, we destroy the `BasisGenerator` and construct a new one for the next window, never using the interval concept.